### PR TITLE
SCM: 54851: Tree view component

### PR DESCRIPTION
**Pull request by SuyashSingh15**

### CODE REVIEW

File: src/lib/components/Form/fields/treeCheckBox.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+export default function treeCheckBox({ column, field, fieldLabel, formik, lookups, data, otherProps, model, fieldConfigs, mode }) {
+  const options = lookups ? lookups[column?.lookup] : [];
+  const tree = buildTree(options);
+  let inputValue = formik.values[field]?.length ? formik.values[field].split(", ") : [];
+  let isDisabled;
+  if (mode !== 'copy') {
+    isDisabled = fieldConfigs?.disabled;
+  }
+  const handleChange = (event, newValue) => {
```
[the remark this answers is not in the document]
Comment (manojpandeyDev18) [reply src/lib/components/Form/fields/treeCheckBox.js:37]: Formating missing

File: src/lib/components/Form/fields/treeCheckBox.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { SimpleTreeView } from '@mui/x-tree-view/SimpleTreeView';
+import { TreeItem } from '@mui/x-tree-view/TreeItem';
+
+const buildTree = (data) => {
+  if (!data) return {};
+  const tree = {};
+  data.forEach((item) => {
```
[the remark this answers is not in the document]
Comment (SuyashSingh15) [reply src/lib/components/Form/fields/treeCheckBox.js:9]: js doc added, and !data handled the basecase, for handling empty array it should work fine using the below code.

File: src/lib/components/Form/fields/treeCheckBox.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          value: item.value.toString(),
+          label: item.label
+        }]
+      }
+    }
+  });
+  return Object.values(tree);
+};
+
+export default function treeCheckBox({ column, field, fieldLabel, formik, lookups, data, otherProps, model, fieldConfigs, mode }) {
```
[the remark this answers is not in the document]
Comment (SuyashSingh15) [reply src/lib/components/Form/fields/treeCheckBox.js:29]: its the lookups that are fetched from db and is an object

File: src/lib/components/Form/fields/treeCheckBox.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (SuyashSingh15) [reply src/lib/components/Form/fields/treeCheckBox.js:37]: we cant assign default values as it wont reflect correctly with formik state.

**Review by manojpandeyDev18 — APPROVED** [review]

**Pull request merged**